DEVELOPER.md

```diff
@@ -0,0 +1,373 @@
+```
+This step uses the Makefile targets for building the controller image, the duplicator resource agent image, and the example test agent image. The images then need to be tagged so that the cluster uses loaded images instead of ones from external sources.
+
+### Install the TestSys cli
+Install `cli` from the repo. 
+*This step can be skipped if the cli is already installed.*
+```bash
+cargo install --path cli
+```
+
+### Create the resource and test yaml file
+The yaml file describes 2 CRDs that will be created.
+`cli run file` has the ability to create as many TestSys objects at the same time.
+Each object should be separated by `---`.
+
+The first is the duplicator resource. 
+This resource takes whatever configuration is provided and returns it as a created resource.
+In this example `{info: 3}` is the provided configuration.
+
+The test agent is very similar to the previous example except we have added a depended resource.
+The resources field of the test CRD signifies that the test should not be run until all resources have finished.
+The other change is the hello count. 
+In the previous example, it was set to be 3.
+In this example, the value is set based on the duplicator resources output.
+Any field can be configured to take values from other resources by using `{<resource name>.<field name>}`.
+```bash
+echo '---
+apiVersion: testsys.bottlerocket.aws/v1
+kind: Resource
+metadata:
+  name: duplicator
+  namespace: testsys-bottlerocket-aws
+spec:
+  agent:
+    name: dup-agent
+    image: "duplicator-resource-agent:demo"
+    keepRunning: false
+    configuration:
+      info: 3
+  dependsOn: []
+---
+apiVersion: testsys.bottlerocket.aws/v1
+kind: Test
+metadata:
+  name: hello-bones
+  namespace: testsys-bottlerocket-aws
+spec:
+  agent:
+    name: hello-agent
+    image: "example-test-agent:demo"
+    keepRunning: false
+    configuration:
+      mode: Fast
+      person: Bones the Cat
+      helloCount: {duplicator.info}
+      helloDurationMilliseconds: 500
+  resources: [duplicator]
+  dependsOn: []' > example.yaml
+```
+
+### Create the `kind` cluster
+Check if the `kind` cluster already exists.
+```bash
+kind get clusters
+```
+
+If the cluster already exists, it should be deleted.
+```bash
+kind delete cluster --name testsys-demo
+```
+
+Create the new demo cluster.
+```bash
+kind create cluster --name testsys-demo
+```
+
+Now the images created earlier need to be added to the cluster.
+```bash
+kind load docker-image controller:demo example-test-agent:demo duplicator-resource-agent:demo --name testsys-demo
+```
+
+### Install TestSys
+Before any tests can be run, the K8s cluster needs to understand what a test is and how to handle them. 
+To do this TestSys needs to be installed to the cluster.
+```bash
+cli install --controller-uri controller:demo
+```
+
+### Run the test
+The CLI installed earlier gives the ability to run tests on a cluster.
+```bash
+cli run file example.yaml
```

Comment:
When running this portion, I get the following error:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Client(ClientError(Deserialization { source: Error("invalid type: map, expected u32", line: 0, column: 0) }))', agent/test-agent/examples/example_test_agent/main.rs:103:10
```

Is the `duplicator.info` selector valid? Maybe something good to add also would be some kind of "Troubleshooting" section of what to look at in the cluster to try and debug.

Reply:
The correct templating was `${duplicator.info}` instead of `{duplicator.info}`.